### PR TITLE
ci: Adding node-linux-medium executor

### LINKED ISSUE #25229: Reduce the size of ressources dedicated to circle.ci job that builds the Extension
### What is this about?

Switching from Docker Medium+ (Medium+, 3 CPU, 6 GB RAM, 15 credits/min) to Linux Medium (2 CPU, 7.5 GB RAM, 10 credits/min) for the build steps (and any step that doesn't require a browser)

Please refer the below ticket:-
https://github.com/MetaMask/metamask-extension/blob/c3875b3601f86677e38c9cde273e09e53dd7abc8/.circleci/config.yml#L16-L21

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by vthomas13**
## **Description**
This PR implements a suggestion from @davidmurdoch to use linux machine image to reduce CI costs on the build steps in our pipeline. Starting with the build steps, we want to optimize for cost for any tests that can support a linux executor.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25476?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/25229

## **Manual testing steps**

1. Check CI

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by davidmurdoch — DISMISSED** [review]
🚀  We'll need to follow up on this PR in a week or so to measure credit usage to be sure we are saving $$$$$s

**Review by DDDDDanica — DISMISSED** [review]

**Review by hjetpoluru — DISMISSED** [review]
LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,12 @@ executors:
     resource_class: medium
     environment:
       NODE_OPTIONS: --max_old_space_size=3072
+  node-linux-medium:
+    machine:
+      image: ubuntu-2204:2024.04.4
```
Comment (HowardBraham) [inline .circleci/config.yml:18]: We're in an Ubuntu LTS transition period.  I think we might as well go with the next version, which is
`ubuntu-2404:2024.05.1`

**Review by HowardBraham — CHANGES_REQUESTED** [review]
This **_should_** reduce credits per workflow, since Linux VM medium costs 10 credits per minute, and Docker Medium+ costs 15 credits per minute.  As long as we don't somehow expand our minutes by 50%.
https://circleci.com/product/features/resource-classes/

There are a few more jobs that I noticed can use `node-linux-medium`:
- prep-build
- prep-build-mv2
- prep-build-mmi
- prep-build-flask
- prep-build-flask-mv2
- prep-build-test-mmi-playwright

**Review by HowardBraham — APPROVED** [review]

**Review by davidmurdoch — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -13,6 +13,12 @@ executors:
     resource_class: medium
     environment:
       NODE_OPTIONS: --max_old_space_size=3072
+  node-linux-medium:
+    machine:
+      image: ubuntu-2404:2024.05.1
+    resource_class: medium #// linux medium: 2 CPUs, 7.5 GB RAM, 10 credits/min
+    environment:
+      NODE_OPTIONS: --max_old_space_size=6144
```
Comment (Gudahtt) [inline .circleci/config.yml:21]: Curious that we're allowing for a larger heap here than on `medium+`. I guess there should be adequate room though, as long as we're not running multiple Node.js processes in one job.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**